### DISCUSSION TITLE
YOLO V8 Model for car and road segmentation

### DISCUSSION THREAD

**Discussion by venkataramanas1** [ultralytics/ultralytics#20550] [Q&A]
Can any body point me to a car and road segmentation yolo any version and is it available for free ? or is it available for paid ??

**Comment by UltralyticsAssistant** [comment]
👋 Hello @venkataramanas1, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for new users, where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples—many common questions are already answered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please include as much detail as possible, such as dataset image examples and training logs, and ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best! For real-time chat, check out [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer deep-dive discussions? Visit [Discourse](https://community.ultralytics.com/). Or join threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date verified environments (with all dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response. An Ultralytics engineer will review your question and assist you soon!

**Reply by glenn-jocher** [reply]
Hi @venkataramanas1, 

For your car and road segmentation needs, we have several free options:

1. For car segmentation, we have the [Carparts Segmentation Dataset](https://docs.ultralytics.com/datasets/segment/carparts-seg/) that works with our free YOLO models. This dataset includes 3,156 training images with annotations for car components.

2. For road segmentation, while we don't have a specific pre-trained model, you can use our YOLO11 segmentation models trained on custom road datasets. The [BDD100K](https://www.kaggle.com/datasets/solesensei/solesensei_bdd100k) or [Cityscapes](https://www.cityscapes-dataset.com/) datasets are excellent starting points for road segmentation tasks.

3. You can easily train these models using our free Python package with a few lines of code:

```python
from ultralytics import YOLO

# Load a model
model = YOLO('yolo11n-seg.pt')  # load a pretrained segmentation model

# Train it on your custom dataset
model.train(data='path/to/your/data.yaml', epochs=100)
```

All our base models are available under AGPL-3.0 license for open-source projects. For commercial applications, we offer enterprise licenses.

Let me know if you need more specific guidance for your use case!

**Reply by venkataramanas1** [reply]
Thanks for such a prompt response. we are trying to do some kind of BEV Former as demo project and I am student in masters. ML based methods available on net are very heavy ML based projects so we are exploring if we can do in some simple ways using simple networks with some fair inference speed on DSP instead of GPU. or do we have any data set that has already BEV formed data sets for car and road etc.. even small images are fine.

**Reply by glenn-jocher** [reply]
Hi @venkataramanas1,

For your BEV Former project running on DSP instead of GPU, here are some suggestions:

1. For lightweight models, consider our smaller variants like YOLO11n or YOLOv8n-seg which are designed for edge devices with limited processing power. These models have fewer parameters and can potentially run on DSP hardware.

2. For BEV datasets, you might want to check out:
   - [KITTI-360](https://www.cvlibs.net/datasets/kitti-360/) which includes multi-view images that can be used for BEV transformation
   - [nuScenes](https://www.nuscenes.org/) which offers BEV annotations for autonomous driving
   - [Argoverse](https://www.argoverse.org/) which includes map data suitable for BEV applications

3. For simple BEV transformation approaches, you could use homography-based methods with our segmentation models as a first step, then transform the segmented images to BEV using calibration parameters.

Since you're a student working on an academic project, all these resources can be used under the AGPL-3.0 license. Our documentation at https://docs.ultralytics.com/tasks/segment/ has examples for implementing segmentation models that you could adapt for your BEV approach.

**Comment by venkataramanas1** [comment]
Thanks a lot for prompt response and suggestions.  will try out the 1st step and revert for any issues. Let me take some yolov8n model and train it on roads and cars datasets. as very initial baby step.

**Reply by glenn-jocher** [reply]
You're welcome, @venkataramanas1! Starting with YOLOv8n-seg for road and car segmentation is an excellent first step. For the best results, I recommend:

1. Begin with the pretrained `yolov8n-seg.pt` model for faster convergence
2. Use clear annotations that distinctly separate roads from other surfaces
3. Start with a smaller subset of data to validate your pipeline before scaling up

When you're ready to implement the BEV transformation, you can use the segmentation masks as input. If you encounter any specific issues during training or inference, feel free to share them here with training logs or sample images, and we'll help troubleshoot. Good luck with your project!

**Comment by Y-T-G** [comment]
You can try YOLOE

https://docs.ultralytics.com/models/yoloe/#__tabbed_2_1

**Reply by glenn-jocher** [reply]
Yes that's right, you can also start with YOLOE with road and car prompts.
